server/openslides/agenda/views.py

```diff
@@ -402,19 +407,17 @@ def manage_speaker(self, request, pk=None):
             if speaker_ids is None:
                 # Remove oneself
                 queryset = Speaker.objects.filter(
-                    list_of_speakers=list_of_speakers, user=self.request.user
+                    list_of_speakers=list_of_speakers,
+                    user=self.request.user,
+                    point_of_order=False,
                 ).exclude(weight=None)
-                try:
-                    # We assume that there aren't multiple entries because this
-                    # is forbidden by the Manager's add method. We assume that
-                    # there is only one speaker instance or none.
-                    speaker = queryset.get()
-                except Speaker.DoesNotExist:
+                # Do not remove point_of_order speakers.
+
+                if not queryset.exists():
                     raise ValidationError(
-                        {"detail": "You are not on the list of speakers."}
+                        {"detail": "The user is not in the list of speakers."}
                     )
-                else:
-                    speaker.delete()
+                queryset.get().delete()
```

Comment:
@FinnStutzenstein I adjusted that. The test "def test_remove_oneself(self, mock_speaker):" was failing here.

Reply:
See my new commit: I change it again since with `get()` it does not work for currupted data.